src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -643,3 +643,130 @@ Profile CascadiaSettings::_CreateDefaultProfile(const std::wstring_view name)
 
     return newProfile;
 }
+
+// Method Description:
+// - Gets our list of warnings we found during loading. These are things that we
+//   knew were bad when we called `_ValidateSettings` last.
+// Return Value:
+// - a reference to our list of warnings.
+std::vector<TerminalApp::SettingsLoadWarnings>& CascadiaSettings::GetLoadWarnings()
+{
+    return _warnings;
+}
+
+// Method Description:
+// - Attempts to validate this settings structure. If there are critical errors
+//   found, they'll be thrown as a SettingsLoadError. Non-critical errors, such
+//   as not finding the default profile, will only result in an error. We'll add
+//   all these warnings to our list of warnings, and the application can chose
+//   to display these to the user.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void CascadiaSettings::_ValidateSettings()
+{
+    _warnings.clear();
+
+    _ValidateProfilesExist();
+    _ValidateNoDuplicateProfiles();
+    _ValidateDefaultProfileExists();
+}
+
+// Method Description:
+// - Checks if the settings contain profiles at all. As we'll need to have some
+//   profiles at all, we'll throw an error if there aren't any profiles.
+void CascadiaSettings::_ValidateProfilesExist()
+{
+    const bool hasProfiles = _profiles.size() != 0;
+    if (!hasProfiles)
+    {
+        // Throw an exception. This is an invalid state, and we want the app to
+        // be able to gracefully use the default settings.
+
+        // We can't add the warning to the list of warnings here, because this
+        // object is not going to be returned at any point.
+
+        throw ::TerminalApp::TerminalException(::TerminalApp::SettingsLoadErrors::NoProfiles);
+    }
+}
+
+// Method Description:
+// - Checks if the "globals.defaultProfile" is set to one of the profiles we
+//   actually have. If the value is unset, or the value is set to something that
+//   doesn't exist in the list of profiles, we'll arbitrarily pick the first
+//   profile to use temporarily as the default.
+// - Appends a SettingsLoadWarnings::MissingDefaultProfile to our list of
+//   warnings if we failed to find the default.
+void CascadiaSettings::_ValidateDefaultProfileExists()
+{
+    const auto defaultProfileGuid = GlobalSettings().GetDefaultProfile();
+    const bool nullDefaultProfile = defaultProfileGuid == GUID{};
+    bool foundDefaultProfile = false;
+    for (const auto& profile : _profiles)
+    {
+        if (profile.GetGuid() == defaultProfileGuid)
+        {
+            foundDefaultProfile = true;
+            break;
+        }
+    }
+
+    const bool defaultProfileNotInProfiles = !foundDefaultProfile;

```

Comment:
We could consolidate `foundDefaultProfile` and `defaultProfileNotInProfiles`. Just start with `defaultProfileNotInProfiles` being `true` in line 705, set it to `false` in line 710. We only use it here anyways.